### PR TITLE
bugfix: skip teardown network, if the sandbox has been stopped

### LINKED ISSUE #1536: [bug] stop sandbox failed for not finding network namespace path
### Ⅰ. Issue Description

When I'm doing node-e2e test for Pouch CRI, there are lot of log like:
```
ERRO[2018-06-15 06:57:18.160883152] Error deleting network: failed to Statfs "/proc/10435/ns/net": no such file or directory 
ERRO[2018-06-15 06:57:18.160936276] StopPodSandbox for "0ad2446f4a41ad160ac10b59a3efeb81a284a955decbee74cb37750d5a303073" failed, error: failed to destroy network for sandbox "0ad2446f4a41ad160ac10b59a3efeb81a284a955decbee74cb37750d5a303073": failed to Statfs "/proc/10435/ns/net": no such file or directory 
```

That means the pod can't be stopped because of the pod network can't be teared down.

However through `pouch ps -a`, the corresponding container has been stopped, so we can't find the corresponding network namespace obviously and we won't stop the sandbox successfully forever.

So I need to configure now whether kubelet will stop sandbox multiple times, if it is true, we need to change the logic of `StopSandbox` in CRI manager.

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### Ⅰ. Describe what this PR did

Please refer: #1536

When stop the sandbox, if the network namespace file of the sandbox does not exist, skip tearing down the pod network.

### Ⅱ. Does this pull request fix one issue?

fixes #1536

### CODE REVIEW

File: cri/v1alpha1/cri.go [hunk 1]
```diff
@@ -284,16 +284,23 @@ func (c *CriManager) StopPodSandbox(ctx context.Context, r *runtime.StopPodSandb
...
+				Namespace:    metadata.GetNamespace(),
+				ID:           podSandboxID,
+				NetNS:        sandboxMeta.NetNSPath,
+				PortMappings: toCNIPortMappings(sandboxMeta.Config.GetPortMappings()),
+			})
+			if err != nil {
+				return nil, err
+			}
+		} else if !os.IsNotExist(err) {
 			return nil, err
```
Comment (fuweid) [inline cri/v1alpha1/cri.go]: In order to add more information in the error, could we use `pkgerrors.Wrap` to wrap the error? WDYT?

File: cri/v1alpha1/cri.go [hunk 2]
```diff
@@ -284,16 +284,23 @@ func (c *CriManager) StopPodSandbox(ctx context.Context, r *runtime.StopPodSandb
...
 		}
+		// If the sandbox has been closed, the corresponding network namespace
```
Comment (fuweid) [inline cri/v1alpha1/cri.go]: Could we put the comment on the top?

File: cri/v1alpha2/cri.go [hunk 3]
```diff
@@ -284,16 +284,23 @@ func (c *CriManager) StopPodSandbox(ctx context.Context, r *runtime.StopPodSandb
...
+				NetNS:        sandboxMeta.NetNSPath,
+				PortMappings: toCNIPortMappings(sandboxMeta.Config.GetPortMappings()),
+			})
+			if err != nil {
+				return nil, err
+			}
+		} else if !os.IsNotExist(err) {
 			return nil, err
 		}
+		// If the sandbox has been closed, the corresponding network namespace
```
Comment (fuweid) [inline cri/v1alpha2/cri.go]: Could we put the comment on the top of `if`? if not, it seems that we delete the code but not comment. 😄

File: cri/v1alpha2/cri.go [hunk 4]
```diff
@@ -284,16 +284,23 @@ func (c *CriManager) StopPodSandbox(ctx context.Context, r *runtime.StopPodSandb
...
+				Namespace:    metadata.GetNamespace(),
+				ID:           podSandboxID,
+				NetNS:        sandboxMeta.NetNSPath,
+				PortMappings: toCNIPortMappings(sandboxMeta.Config.GetPortMappings()),
+			})
+			if err != nil {
+				return nil, err
+			}
+		} else if !os.IsNotExist(err) {
 			return nil, err
```
Comment (fuweid) [inline cri/v1alpha2/cri.go]: use `pkgerrors.Wrap` to show the error which means that the file has been removed. WDYT?

**Comment by YaoZengzeng** [comment]
@fuweid Updated.

**Comment by fuweid** [comment]
LGTM

**Pull request merged**